docs/4.x/tracking-javascript.md

```diff
@@ -109,13 +109,24 @@ Matomo provides a mechanism to manage your user's tracking consent. You can requ
 *   `setConsentGiven()` - Mark that the current user has consented. The consent is one-time only, so in a subsequent browser session, the user will have to consent again. To remember consent, see the method below: `rememberConsentGiven`.
 *   `rememberConsentGiven( hoursToExpire )` - Mark that the current user has consented, and remembers this consent through a browser cookie. The next time the user visits the site, Matomo will remember that they consented, and track them. If you call this method, you do not need to call `setConsentGiven`.
 *   `forgetConsentGiven()` - Remove a user's consent, both if the consent was one-time only and if the consent was remembered. After calling this method, the user will have to consent again in order to be tracked.
+*   `hasRememberedConsent()` - Returns true or false depending on whether the current visitor has given consent previously or not.
+*   `getRememberedConsent()` - If consent was given, returns the timestamp when the visitor gave consent. Only works if `rememberConsentGiven` was used and not when `setConsentGiven` was used. The timestamp is the local timestamp which depends on the visitors time.
+*   `isConsentRequired()` - Returns true or false depending on whether `requireConsent` was called previously.
 
 Matomo also provides a mechanism to manage your user's cookie consent. You can require that users consent to using cookies. Tracking requests will be always sent but depending on the consent cookies will be used or not used.
 
 *   `requireCookieConsent()` - By default the Matomo tracker assumes consent to using cookies. To change this behavior so no cookies are used by default, you must call `requireCookieConsent`.
 *   `setCookieConsentGiven()` - Mark that the current user has consented to using cookies. The consent is one-time only, so in a subsequent browser session, the user will have to consent again. To remember cookie consent, see the method below: `rememberCookieConsentGiven`.
 *   `rememberCookieConsentGiven( hoursToExpire )` - Mark that the current user has consented to using cookies, and remembers this consent through a browser cookie. The next time the user visits the site, Matomo will remember that they consented, and use cookies. If you call this method, you do not need to call `setCookieConsentGiven`.
 *   `forgetCookieConsentGiven()` - Remove a user's cookie consent, both if the consent was one-time only and if the consent was remembered. After calling this method, the user will have to consent again in order for cookies to be used.
+*   `areCookiesEnabled()` - Returns true or false depending on whether cookies are currently enabled or disabled.
+
+### Managing opt out
+Wanting to build a custom opt-out form instead of a consent screen or instead of using our [opt-out iframe](https://matomo.org/faq/general/faq_20000/)? Check out the guide for [creating a custom opt-out form](https://developer.matomo.org/guides/tracking-javascript-guide#optional-creating-a-custom-opt-out-form).
```

Comment:
```suggestion
Do you want to build a custom opt-out form instead of a consent screen or instead of using our [opt-out iframe](https://matomo.org/faq/general/faq_20000/)? Check out the guide for [creating a custom opt-out form](https://developer.matomo.org/guides/tracking-javascript-guide#optional-creating-a-custom-opt-out-form).
```

Just to simplify the sentence a bit